roop/core.py

```diff
@@ -135,13 +135,17 @@ def preview_video(video_path):
 
 
 def select_face():
-    args['source_img'] = filedialog.askopenfilename(title="Select a face")
-    preview_image(args['source_img'])
+    source_img = filedialog.askopenfilename(title="Select a face")
+    if source_img and source_img != "" and source_img != args['source_img']:
+        args['source_img'] = source_img
+        preview_image(args['source_img'])
```

Comment:
Really small nitpick, but it would probably be better to pass `source_img` as a parameter instead of `args['source_img']`, just so Python doesn't have to do an extra operation to get the dictionary value. It's still O(1) though so again, really small nitpick and makes no real difference.